src/pyqasm/visitor.py

```diff
@@ -88,6 +88,7 @@ def __init__(
         self._unroll_barriers: bool = unroll_barriers
         self._curr_scope: int = 0
         self._label_scope_level: dict[int, set] = {self._curr_scope: set()}
+        self._recording_depth = True
```

Comment:
Let's name this to `self._recording_ext_gate_depth` as `recording_depth` sounds very general.
Then, `True` will mean we are inside the external gate and `False` otherwise. Essentially, we just flip your current logic